compiler/dcalc/typing.ml

```diff
@@ -490,16 +509,25 @@ and typecheck_expr_top_down
     Bindlib.box @@ unify_and_mark e1 (unionfind_make (TLit TUnit))
   | A.ELit LEmptyError as e1 ->
     Bindlib.box @@ unify_and_mark e1 (unionfind_make (TAny (Any.fresh ())))
-  | A.ETuple (es, s) ->
+  | A.ETuple (es, None) ->
```

Comment:
Why didn't you change to distinct `ETuple` and `EStruct` here too?

Reply:
Yes, that'll probably be better — but I'm trying to not do too much at once, these patches are already getting huge :)